salt/utils/win_lgpo_auditpol.py

```diff
@@ -298,8 +288,176 @@ def get_auditpol_dump():
+
+        refresh (bool):
+            Refresh secedit data stored in __context__. This is needed for
+            testing where the state is setting the value, but the module that
+            is checking the value has its own __context__.
+
+    Returns:
+        bool: ``True`` if successful, otherwise ``False``
+    """
+    if refresh is True:
+        _get_advaudit_values.cache_clear()
+    return _get_advaudit_values().get(option, None)
+
+
+def advaudit_check_csv():
+    """
+    This function checks for the existence of the `audit.csv` file here:
+    `C:\\Windows\\security\\audit`
+
+    If the file does not exist, then it copies the `audit.csv` file from the
+    Group Policy location:
+    `C:\\Windows\\System32\\GroupPolicy\\Machine\\Microsoft\\Windows NT\\Audit`
+
+    If there is no `audit.csv` in either location, then a default `audit.csv`
+    file is created.
+    """
+    system_root = os.environ.get("SystemRoot", "C:\\Windows")
+    f_audit = pathlib.Path(system_root, "security", "audit", "audit.csv")
+    f_audit_gpo = pathlib.Path(
+        system_root,
+        "System32",
+        "GroupPolicy",
+        "Machine",
+        "Microsoft",
+        "Windows NT",
+        "Audit",
+        "audit.csv",
+    )
+    # Make sure there is an existing audit.csv file on the machine
+    if not f_audit.exists():
+        if f_audit_gpo.exists():
+            # If the GPO audit.csv exists, we'll use that one
+            shutil.copyfile(str(f_audit_gpo), str(f_audit))
+        else:
+            field_names = get_advaudit_defaults("fieldnames")
+            # If the file doesn't exist anywhere, create it with default
+            # fieldnames
+            f_audit.parent.mkdir(parents=True, exist_ok=True)
+            f_audit.write_text(",".join(field_names))
+
+
+@lru_cache
+def _get_advaudit_defaults():
+    # Get available setting names and GUIDs
+    # This is used to get the fieldnames and GUIDs for individual policies
+    log.debug("Loading auditpol defaults into __context__")
+    dump = salt.utils.win_lgpo_auditpol.get_auditpol_dump()
+    reader = csv.DictReader(dump)
+    audit_defaults = {"fieldnames": reader.fieldnames}
+    for row in reader:
+        row["Machine Name"] = ""
+        row["Auditpol Name"] = row["Subcategory"]
+        # Special handling for snowflake scenarios where the audit.csv names
+        # don't match the auditpol names
+        if row["Subcategory"] == "Central Policy Staging":
+            row["Subcategory"] = "Audit Central Access Policy Staging"
+        elif row["Subcategory"] == "Plug and Play Events":
+            row["Subcategory"] = "Audit PNP Activity"
+        elif row["Subcategory"] == "Token Right Adjusted Events":
+            row["Subcategory"] = "Audit Token Right Adjusted"
+        else:
+            row["Subcategory"] = "Audit {}".format(row["Subcategory"])
+        audit_defaults[row["Subcategory"]] = row
+    return audit_defaults
+
+
+def get_advaudit_defaults(option=None):
```

Comment:
same here, i'm not following why we are adding these new functions?

Reply:
So we can run tests against the functionality

Reply:
Wait, this one is weird, let me check

Reply:
This was a private utility function in the execution module just moved to the utility module.